hazelcast-jet-core/src/main/java/com/hazelcast/jet/accumulator/LinRegAccumulator.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.accumulator;
+
+import com.hazelcast.jet.JetException;
+import com.hazelcast.nio.ObjectDataOutput;
+
+import java.io.IOException;
+import java.math.BigInteger;
+
+/**
+ * Maintains the components needed to compute the linear regression on a
+ * set of {@code (long, long)} pairs. The intermediate results are held in
+ * the form of {@link BigInteger} and the finished value is a {@code
+ * double}-valued linear coefficient.
+ */
+public final class LinRegAccumulator {
```

Comment:
I'm fine with the current name... if we ever need to differentiate, we'll rename to more specific names.